### PR TITLE
Refs #23919 -- Replaced super(ClassName, self) with super()

### PR DESCRIPTION
**Pull request by ChillarAnand**

### CODE REVIEW

File: django/core/serializers/json.py [hunk 1]
```diff
@@ -62,7 +62,7 @@ def end_object(self, obj):
 
     def getvalue(self):
         # Grand-parent super
-        return super(PythonSerializer, self).getvalue()
+        return super().getvalue()
```
Comment (charettes) [inline django/core/serializers/json.py]: This is an occurence of what @aaugustin mentioned in #7875 and why this change cannot be automated by a simple search and replace.

Comment (ChillarAnand) [inline django/core/serializers/json.py]: Instead of blind search/replace, I am doing a [replace by class name](https://github.com/ChillarAnand/01/blob/master/python/super_exp.py). Here are sample [input](https://github.com/ChillarAnand/01/blob/master/python/super_old.py) & [output](https://github.com/ChillarAnand/01/blob/master/python/super_new.py). Will that work? @aaugustin

Comment (aaugustin) [inline django/core/serializers/json.py]: Looking at your script, the intent seems correct, but the fact that this code was changed incorrectly proves that the implementation doesn't work (yet).

Comment (ChillarAnand) [inline django/core/serializers/json.py]: This happened when I replaced it using `sed`. Now I have passed codebase throught that script and it looks fine now?

File: django/contrib/staticfiles/storage.py [hunk 2]
```diff
@@ -403,7 +403,7 @@ def load_manifest(self):
 
     def post_process(self, *args, **kwargs):
         self.hashed_files = OrderedDict()
-        all_post_processed = super(ManifestFilesMixin,
+        all_post_processed = super(,
```
Comment (timgraham) [inline django/contrib/staticfiles/storage.py]: It seems you'll need to check the patch by hand. This is an obvious syntax error.

File: django/core/cache/backends/memcached.py [hunk 3]
```diff
@@ -154,7 +154,7 @@ class MemcachedCache(BaseMemcachedCache):
     "An implementation of a cache binding using python-memcached"
     def __init__(self, server, params):
         import memcache
-        super(MemcachedCache, self).__init__(server, params,
+        super().__init__(server, params,
```
Comment (timgraham) [inline django/core/cache/backends/memcached.py]: And also fix formatting in cases like this.  Use hanging indent::
```
super().__init__(
    server, param, library=pylibmc, value_not_found_exception=pylibmc.NotFound,
)
```
or a single line if it fits in 119 characters.

**Comment by timgraham** [comment]
Also, please use rebase so there isn't a merge commit in your branch.

**Comment by ChillarAnand** [comment]
@timgraham Rebased. Thanks.

**Comment by timgraham** [comment]
I reviewed the patch and force pushed a few cosmetic updates, however, this doesn't appear complete. A quick look at `grep -rI "super(" *` shows places that aren't updated. It looks like you missed `super(..., cls)` and also `super()` in nested classes, e.g. classes in test methods.

Please push updates in a separate commit so I don't have to review the entire patch again.

**Comment by ChillarAnand** [comment]
Total super calls
```
→ grep -rI "super(" | wc -l         
   1364   
```

All of them were replaced except a few where there is some hacky stuff is going on or there are calls to grand parents.

```
→ grep -rI "super(" | grep -v 'super()' | wc -l
     31     
```

I have also replaced super in docs using `sed` which is in separate commit. Will squash all commits in the end.
